pkg/kubelet/volume/volume_manager.go

```diff
@@ -0,0 +1,349 @@
+	controllerAttachDetachEnabled bool,
+	hostName string,
+	podManager pod.Manager,
+	kubeClient internalclientset.Interface,
+	volumePluginMgr *volume.VolumePluginMgr) (VolumeManager, error) {
+	vm := &volumeManager{
+		kubeClient:          kubeClient,
+		volumePluginMgr:     volumePluginMgr,
+		desiredStateOfWorld: cache.NewDesiredStateOfWorld(volumePluginMgr),
+		actualStateOfWorld:  cache.NewActualStateOfWorld(hostName, volumePluginMgr),
+		operationExecutor:   operationexecutor.NewOperationExecutor(volumePluginMgr),
+	}
+
+	vm.reconciler = reconciler.NewReconciler(
+		controllerAttachDetachEnabled,
+		reconcilerLoopSleepPeriod,
+		waitForAttachTimeout,
+		hostName,
+		vm.desiredStateOfWorld,
+		vm.actualStateOfWorld,
+		vm.operationExecutor)
+	vm.desiredStateOfWorldPopulator = populator.NewDesiredStateOfWorldPopulator(
+		kubeClient,
+		desiredStateOfWorldPopulatorLoopSleepPeriod,
+		podManager,
+		vm.desiredStateOfWorld)
+
+	return vm, nil
+}
+
+type volumeManager struct {
+	// kubeClient is the kube API client used by DesiredStateOfWorldPopulator to
+	// communicate with the API server to fetch PV and PVC objects
+	kubeClient internalclientset.Interface
+
+	// volumePluginMgr is the volume plugin manager used to access volume
+	// plugins. It must be pre-initiazlied.
+	volumePluginMgr *volume.VolumePluginMgr
+
+	// desiredStateOfWorld is a data structure containing the desired state of
+	// the world according to the manager: i.e. what volumes it wants be
+	// attached and which pods are referencing the volumes).
+	// The data structure is populated by the desired state of the world
+	// populator.
+	desiredStateOfWorld cache.DesiredStateOfWorld
+
+	// actualStateOfWorld is a data structure containing the actual state of
+	// the world according to the manager: i.e. which volumes are attached to
+	// this node and what pods the volumes are mounted to.
+	// The data structure is populated upon successful completion of attach,
+	// detach, mount, and unmount actions triggered by the reconciler.
+	actualStateOfWorld cache.ActualStateOfWorld
+
+	// operationExecutor is used to start asynchronous attach, detach, mount,
+	// and unmount operations.
+	operationExecutor operationexecutor.OperationExecutor
+
+	// reconciler runs an asynchronous periodic loop to reconcile the
+	// desiredStateOfWorld with the actualStateOfWorld by triggering attach,
+	// detach, mount, and unmount operations using the operationExecutor.
+	reconciler reconciler.Reconciler
+
+	// desiredStateOfWorldPopulator  runs an asynchronous periodic loop to
+	// populate the desiredStateOfWorld using the kublet PodManager.
+	desiredStateOfWorldPopulator populator.DesiredStateOfWorldPopulator
+}
+
+func (vm *volumeManager) Run(stopCh <-chan struct{}) {
+	defer runtime.HandleCrash()
+	glog.Infof("Starting Kubelet Volume Manager")
+
+	go vm.reconciler.Run(stopCh)
+	go vm.desiredStateOfWorldPopulator.Run(stopCh)
+
+	<-stopCh
+	glog.Infof("Shutting down Kubelet Volume Manager")
```

Comment:
Hmm, why? They should only happen once per binary start/stop--not particularly verbose. And good debug info that the component even started